### PR TITLE
Regex replace

### PR DESCRIPTION
**Pull request by jfolk2015**
## Reminders:

- [x] If changing elixir code in an "app", did you update the relevant version
      in `mix.exs`?
- [ ] If altering an API endpoint, was the relevant postman collection updated?

### CODE REVIEW

File: apps/transformers/lib/transformations/regex_replace.ex [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      transformed_value = Regex.replace(regex, value, replacement)
+      transformed_payload = Map.put(payload, source_field, transformed_value)
+      {:ok, transformed_payload}
+    else
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp abort_if_not_string(value, field_name) do
+    if is_bitstring(value) do
```
Comment (christinepoydence) [inline apps/transformers/lib/transformations/regex_replace.ex:25]: I think we might prefer `is_binary` over `is_bitstring`? 

(https://stackoverflow.com/questions/54161232/use-is-bitstring-or-is-binary-in-elixir-guard-for-string)

File: apps/transformers/test/unit/transformations/regex_replace_test.exs [hunk 2]
```diff
@@ -0,0 +1,162 @@
+defmodule Transformers.RegexReplaceTest do
+  use ExUnit.Case
+
+  alias Transformers.RegexReplace
+
+  test "when source field not specified, return error" do
```
Comment (christinepoydence) [inline apps/transformers/test/unit/transformations/regex_replace_test.exs:6]: Do you see any value in using a data test for these three (very similar) unit tests? Something like:

```
data_test "when #{missing_field} field not specified, return error" do
     payload = %{
       "something" => "123"
     }

     parameters = %{
       "regex" => "^(\\w+)",
       "replacement" => "a",
       "source" => "something"
     } |> Map.delete(missing_field)

     {:error, reason} = RegexReplace.transform(payload, parameters)

     assert reason == "Missing transformation parameter: #{error_message}"

     where([
        [:missing_field, :error_message],
        ["source", "sourceField"],
        ["regex", "regex"],
        ["replacement", "replacement"],
     ])
   end
```

Warning: The above is untested code.

**Review by christinepoydence — APPROVED** [review]

File: apps/transformers/lib/transformations/regex_replace.ex (hunk 1 above)
Comment (jfolk2015) [reply apps/transformers/lib/transformations/regex_replace.ex]: :( That's a confusing name and I wish they would just implement "is_string". Fixed.

File: apps/transformers/test/unit/transformations/regex_replace_test.exs (hunk 2 above)
Comment (jfolk2015) [reply apps/transformers/test/unit/transformations/regex_replace_test.exs:6]: After some googling, I'm not finding documentation for data_test. Can you point me in the right direction?

Comment (jfolk2015) [reply apps/transformers/test/unit/transformations/regex_replace_test.exs]: Nevermind, found it on the Chechov library. I'll try it out in this case and we can see how we like it.

**Pull request merged**